processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -23,6 +27,10 @@ type Detector interface {
 	Detect(ctx context.Context) (resource pcommon.Resource, schemaURL string, err error)
 }
 
+type ExposeHandlerDetector interface {
+	Detector // Embed the existing Detector interface
+	ExposeHandlers(ctx context.Context) *request.Handlers
+}
```

Comment:
Don't need to embed the `Detector`. The interface can just have the Handlers function. Doesn't really matter if it's a `Detector` or not.

```suggestion
type HandlerProvider interface {
	Handlers() *request.Handlers
}
```